metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -312,14 +312,25 @@ class Compilers(
   }
 
   def hover(
-      params: TextDocumentPositionParams,
+      params: HoverExtParams,
       token: CancelToken
-  ): Future[Option[Hover]] =
-    withPCAndAdjustLsp(params) { (pc, pos, adjust) =>
-      pc.hover(CompilerOffsetParams.fromPos(pos, token))
-        .asScala
-        .map(_.asScala.map { hover => adjust.adjustHoverResp(hover) })
+  ): Future[Option[Hover]] = {
+    if (params.range != null) {
+      withPCAndAdjustLsp(params) { (pc, pos, adjust) =>
+        pc.hover(CompilerRangeParams.fromPos(pos, token))
+          .asScala
+          .map(_.asScala.map { hover => adjust.adjustHoverResp(hover) })
+      }
+    } else {
+      val param: TextDocumentPositionParams =
+        new TextDocumentPositionParams(params.textDocument, params.position)
+      withPCAndAdjustLsp(param) { (pc, pos, adjust) =>
+        pc.hover(CompilerOffsetParams.fromPos(pos, token))
+          .asScala
+          .map(_.asScala.map { hover => adjust.adjustHoverResp(hover) })
+      }
```

Comment:
There is another difference in `withPCAndAdjustLsp` methods- the second version creates `scala.meta.inputs.Position` from range. This is a bit harder to deduplicate, since `toMeta` requires local value- `input`.
Deduplicated the adjustment part, see if it's ok :smiley: